src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,24 @@
+import { createSlice } from '@reduxjs/toolkit';
+import uuid from 'react-uuid';
+
+const initialState = { books: [] };
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.books.push({
+        bookId: uuid(),
+        name: action.payload.name,
+        author: action.payload.author,
+      });
+    },
+    removeBook: (state, action) => {
+      state.books.filter((book) => book.id !== action.payload.id);
+    },
```

Comment:
You have done a very good job putting all the reducers into the redux store and I must say, you did it very well 👍 However, please note that the filter method returns a new array and hence you have to assign that new array to the state. Along the line you will learn about immutability and get to know a better to do this but for now, you can change line 18 to look something like this: 

```sh
   state.books = state.books.filter((book) => book.id !== action.payload.id);
```

Reply:
resolved! ✔️